sql/src/test/java/io/crate/operation/projectors/FetchProjectorTest.java

```diff
@@ -0,0 +1,190 @@
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.projectors;
+
+import com.carrotsearch.hppc.IntHashSet;
+import com.carrotsearch.hppc.IntObjectHashMap;
+import com.carrotsearch.hppc.IntObjectMap;
+import com.carrotsearch.hppc.IntSet;
+import com.google.common.collect.Iterables;
+import com.google.common.util.concurrent.SettableFuture;
+import io.crate.analyze.symbol.FetchReference;
+import io.crate.analyze.symbol.InputColumn;
+import io.crate.analyze.symbol.Reference;
+import io.crate.analyze.symbol.Symbol;
+import io.crate.core.collections.Bucket;
+import io.crate.core.collections.CollectionBucket;
+import io.crate.core.collections.Row;
+import io.crate.core.collections.Row1;
+import io.crate.metadata.*;
+import io.crate.metadata.table.ColumnPolicy;
+import io.crate.operation.RowUpstream;
+import io.crate.operation.projectors.fetch.FetchProjector;
+import io.crate.operation.projectors.fetch.FetchProjectorContext;
+import io.crate.operation.projectors.fetch.TransportFetchOperation;
+import io.crate.planner.node.fetch.FetchSource;
+import io.crate.test.integration.CrateUnitTest;
+import io.crate.testing.CollectingRowReceiver;
+import io.crate.testing.TestingHelpers;
+import io.crate.types.LongType;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+import static org.hamcrest.core.Is.is;
+import static org.mockito.Mockito.*;
+
+public class FetchProjectorTest extends CrateUnitTest {
+
+    private static final TableIdent USER_TABLE_IDENT = new TableIdent(Schemas.DEFAULT_SCHEMA_NAME, "users");
+    private ThreadPool threadPool;
+
+    @Before
+    public void before() throws Exception {
+        threadPool = new ThreadPool("testThreadPool");
+    }
+
+    @After
+    public void after() throws Exception {
+        threadPool.shutdown();
+        threadPool.awaitTermination(1, TimeUnit.SECONDS);
+    }
+
+    @Test
+    public void testFetchSize() throws Throwable {
+        CollectingRowReceiver rowReceiver = new CollectingRowReceiver();
+
+        TransportFetchOperation transportFetchOperation = mock(TransportFetchOperation.class);
```

Comment:
Would it be possible to implement/mock the interface (FetchOperation) instead of the concrete implementation?